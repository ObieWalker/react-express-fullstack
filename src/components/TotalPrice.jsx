import React from 'react';

const TotalPrice = ({groceries}) => {
  let totalPrice = 0;
  let priceAppended
  groceries.map(grocery => {
    if (grocery.purchased){
      totalPrice += grocery.price
    }
    priceAppended = totalPrice.toFixed(2)
    return priceAppended
  })
  return (
      <div>
        <p>Total Price: &#8358; {priceAppended}</p>
      </div>
    );
  }

export default TotalPrice;
