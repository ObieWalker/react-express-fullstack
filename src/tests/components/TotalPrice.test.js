import React from 'react';
import ReactDOM from 'react-dom';
import expect from 'expect';
import { shallow } from 'enzyme';
import TotalPrice from '../../components/TotalPrice';

let mounted;
let props;
let wrapper;

const getComponent = () => {
  if (!mounted) {
    props = {
      groceries: [{
        purchased: true
      }]
    };
    mounted = shallow(<TotalPrice {...props} />);
  }
  return mounted;
};

describe('TotalPrice Component', () => {
  beforeEach(() => {});

  it('renders component successfully', () => {
    wrapper = getComponent();
    expect(wrapper).toMatchSnapshot();
  });

});

