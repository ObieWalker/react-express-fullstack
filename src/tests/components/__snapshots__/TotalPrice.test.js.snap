// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TotalPrice Component renders component successfully 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TotalPrice
    groceries={
      Array [
        Object {
          "purchased": true,
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <p>
        Total Price: ₦ 
        NaN
      </p>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "Total Price: ₦ ",
          NaN,
        ],
      },
      "ref": null,
      "rendered": Array [
        "Total Price: ₦ ",
        NaN,
      ],
      "type": "p",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <p>
          Total Price: ₦ 
          NaN
        </p>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Total Price: ₦ ",
            NaN,
          ],
        },
        "ref": null,
        "rendered": Array [
          "Total Price: ₦ ",
          NaN,
        ],
        "type": "p",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
